common/extensions.go

```diff
@@ -70,19 +70,30 @@ const SigAzure = azbfs.SigAzure
 const SigXAmzForAws = azbfs.SigXAmzForAws
 
 func RedactSecretQueryParam(rawQuery, queryKeyNeedRedact string) (bool, string) {
-	rawQuery = strings.ToLower(rawQuery) // lowercase the string so we can look for ?[queryKeyNeedRedact] and &[queryKeyNeedRedact]=
+	// make a dictionary of all the params map[param]value
+	copyQuery := rawQuery
+	originalDict := map[string]string{}
+	for _, param := range strings.Split(copyQuery, "&") {
```

Comment:
Check this out: https://pkg.go.dev/net/url#example-ParseQuery
Do you think this will make code efficient and concise?

Reply:
Okay 👍